### ISSUE TITLE
Homebrew console ports

### ISSUE THREAD

**Issue by iCrazyBlaze** [minetest/minetest#12624]
Minetest is currently not available in any way on these platforms as far as I am aware:

- Nintendo Wii, Wii U, or Switch*
- Nintendo 3DS
- PlayStation Vita
- Xbox 360 or any newer Xbox console

\* An old WIP build was made but never finished https://github.com/emmauss/minetest

## Porting problems
Minetest uses a custom engine/backend which will make it harder to port, instead of a standard library like SDL. Iirlicht needs to be ported before any other work can commence.

Controller support will also be a problem for all of these platforms as they use non-standard platform-specific control types more often than not.

## Why port?
Minetest is on basically every PC and mobile OS, so it makes sense to keep expanding its availability, even if just for the novelty or "fun" of doing so. Having a standardised, high quality Minecraft clone on every console is very appealing, and the sheer amount of other content available makes it much more than just that. I would love to see some more discussion of this!

**Comment by TMcSquared** [comment]
I think the main problem with porting would be the amount of time dedicated to maintaining said ports. Just like with the little amount of macOS support, there isn't anything against it, there just isn't a suitable number of people that actively work on that side of things.

**Comment by rollerozxa** [comment]
Irrlicht is not equivalent to SDL, in fact IrrlichtMt even has a new SDL2 device contributed thanks to the Switch port, which will hopefully become the standard device soon alongside controls switching to use SDL instead of Irrlicht.

**Comment by Wuzzy2** [comment]
Suggested new issue title: “Port Minetest to Wii, Wii U, Switch, 3DS, PS Vita and Xbox 360”

**Comment by iCrazyBlaze** [comment]
I am particularly interested in the Wii port but other consoles would be much easier to create. As far as maintaining them goes, as long as the version that the port based on is a stable version with many compatible mods then updates could be made on a non-regular basis. I personally wouldn't mind being stuck with an older version of Minetest and the games available for it for a few weeks, months or even a year, as long as whatever games and mods I am using still work.

In short, these versions wouldn't need to be actively maintained as long as we're willing to sacrifice being on the bleeding-edge of Minetest content,

**Comment by tigercoding56** [comment]
maybe only port the installer to the consoles ( build instructions )

**Comment by rubenwardy** [comment]
Minetest has limited resources. Contributors are welcome to try to port Minetest to their consoles via homebrew, but this isn't something we should track as an issue

**Comment by emmauss** [comment]
I got really far with the port, being able to render the Main Menu and settings. Unfortunately it crashed in lua when trying to load a game.

**Comment by Boombox17556** [comment]
IK i might be an idiot but would it be possible to create a .dol or.elf file that runs a version of the game that's already availiable and emulating any directories needed?

**Comment by Redgabriel13-red13** [comment]
> I got really far with the port, being able to render the Main Menu and settings. Unfortunately it crashed in lua when trying to load a game.

Love2d uses lua and a lot of people ported it for consoles with homebrew stuff
